### ISSUE TITLE
Import and destructure mongoose es6

### ISSUE THREAD

**Issue by hasusozam** [Automattic/mongoose#7009]
I'm using ES6 in my project and I wanted to import mongoose that way:

```
import mongoose from 'mongoose'

const { Schema, model } = mongoose

const bookSchema = new Schema({
  pages: [
    {
      type: Schema.Types.ObjectId,
      ref: 'page'
    }
  ]
})

export default model('book', bookSchema)
```

But I'm getting `TypeError: Cannot read property 'modelSchemas' of undefined`, how can I solve this?

**Comment by Xydan** [comment]
I'm trying to figure it out as well, but it apparently it's a known bug that won't be fixed anytime soon. The are a lot of oddities with mongoose at the higher level, and this is one of them. Here's a response from another user. 

> Supporting this behavior would involve a massive refactor of mongoose connections. Long story short, require('mongoose').model(); is different than var model = require('mongoose').model; model(); because the value of this in the former function call is the require('mongoose'); singleton, whereas it is the global object in the latter call. If you want a workaround, model.call(require('mongoose'), 'MyModelName', schema); should work fine. I'm not a huge fan of the fact that mongoose's top level functions are laden with side effects, but to get rid of that we'd also have to get rid of the mongoose.model() getter syntax, which would break a lot of people's code.

**Comment by hasusozam** [comment]
That is sad but thank you for pointing it out.

**Comment by vkarpov15** [comment]
Yeah unfortunately this issue isn't something we can fix. For example, mongoose.set('debug', true) won't work if you create a model with `import {model} from mongoose`. We will add this to the faq
